src/SignhostAPIClient/Rest/SignHostApiClient.cs

```diff
@@ -26,6 +26,52 @@ public class SignHostApiClient
 		private readonly ISignHostApiClientSettings settings;
 		private readonly HttpClient client;
 
+		public static bool IsPostbackChecksumValid(
+			IDictionary<string, string[]> headers,
+			string body,
+			string sharedSecret,
+			out Transaction postbackTransaction)
+		{
+			postbackTransaction = null;
+			string[] postbackChecksumArray;
+			string postbackChecksum;
+			string calculatedChecksum;
+			Postback postback;
+
+			try {
+				postback = JsonConvert.DeserializeObject<Postback>(body);
+				postbackTransaction = postback;
+			} catch (InvalidCastException e) {
+				return false;
+			} catch {
+				return false;
+			}
+
+			if (headers.ContainsKey("Checksum")) {
+				headers.TryGetValue("Checksum", out postbackChecksumArray);
```

Comment:
This can be combined into one statement, eg
`if (headers.TryGetValue("Checksum", out postbackChecksumArray)) {` ...
Avoids a double lookup in the dictionary, but makes it more terse / readable